src/main/java/org/jabref/gui/texparser/ParseTexDialogViewModel.java

```diff
@@ -117,18 +119,33 @@ public void searchButtonClicked() {
                           noFilesFound.set(true);
                           searchInProgress.set(true);
                           successfulSearch.set(false);
-
                       })
                       .onFinished(() -> searchInProgress.set(false))
                       .onSuccess(newRoot -> {
                           root.set(newRoot);
                           noFilesFound.set(false);
                           successfulSearch.set(true);
                       })
-                      .onFailure(dialogService::showErrorDialogAndWait)
+                      .onFailure(handleFailure())
                       .executeWith(taskExecutor);
     }
 
+    private Consumer<Exception> handleFailure() {
+        return exception -> {
+            root.set(null);
+            noFilesFound.set(true);
+            searchInProgress.set(false);
+            successfulSearch.set(false);
+
+            final boolean permissionProblem = exception instanceof IOException && exception.getCause() instanceof FileSystemException && exception.getCause().getMessage().endsWith("Operation not permitted");
```

Comment:
Does it make sense to move this code to `dialogService.showErrorDialogAndWait(exception)` so that access issues in other parts of JabRef are treated similarly?

Reply:
I wouldn't do that because the `dialogService` is a global thing, whereas how we handle the exception here is quite special.

Reply:
ok, so the same exception but e.g. thrown while opening a bib file should still be displayed in the default error dialog with the stack trace instead of by the specialized permission problem dialog?

Reply:
I am not sure, but I think that in those cases the selection file dialog should show a warning like this. Here the problem is related to the deep search in all subdirectories, if I understood properly.

![Warning](https://user-images.githubusercontent.com/12954316/61490798-0a140a80-a9ae-11e9-83f9-0d2c34875018.png)

Reply:
At least add a complete error message with stack trace to the log. I hate error messages without stack traces.